src/tabs/Tab.js

```diff
@@ -0,0 +1,99 @@
+import PropTypes from 'prop-types';
+import { TabStyles } from './Tabs.styles';
+
+const Tab = (props) => {
+  const {
+    children,
+    activeClassName,
+    className,
+    component,
+    onClick,
+    href,
+    dataTestId,
+    ...rest
+  } = props;
+
+  let as = '';
+  if (href) {
+    as = 'a';
+  } else if (!href && onClick) {
+    as = 'button';
+  }
+
+  let customComponentProps = {};
+  if (as === 'button') {
+    customComponentProps = { type: 'button', ...rest };
+  }
+  if (as === 'a') {
+    customComponentProps = { ...rest };
+  }
+  if (component) {
+    as = component;
+    customComponentProps = { ...rest };
+  }
+
+  return (
+    <TabStyles
+      as={as}
+      onClick={onClick}
+      className={className}
+      href={href}
+      data-testid={dataTestId}
+      $activeClassName={activeClassName}
```

Comment:
@tlouth19  This is a transient prop. Not sure if you have seen it before but the "$" makes it so Styled Components doesn't add it to the DOM and its just used internally. This fixes the warning where the browser doesn't want a camelCase attribute on the DOM.

Reply:
https://stackoverflow.com/questions/57586654/styled-component-attrs-react-does-not-recognize-prop